zeekpkg/manager.py

```diff
@@ -595,6 +598,155 @@ def source_packages(self):
+
+        Returns:
+            str: version with a numeric suffix stripped to be
+            semantic_version compatible.
+
+        """
+        m = re.match(r"([0-9]+\.[0-9]+\.[0-9]+)-[0-9]+", version)
+        if m:
+            return m.group(1)
+
+        return version
+
+    def _discover_spicy_plugin(self, zeek):
+        """
+        Poke zeek -N for Zeek::Spicy and return its version if available.
+
+        There are plans to have spicy-plugin become part of Zeek 6.0 and
+        that means it won't be part of included_plugins anymore. Also, this
+        allows to find a included spicy-plugin plugin for versions which
+        do not have -V/--build-info.
+
+        Returns:
+            str or None: spicy-plugin version as or None if not part of Zeek.
+        """
+        plugin_list = subprocess.check_output([zeek, "-N"], timeout=10)
+        for line in plugin_list.decode("utf-8").splitlines():
+            if line.startswith("Zeek::Spicy") and line.endswith("(built-in)"):
+                # Unfortunately, there's no version output from Zeek when
+                # plugins are built-in. Use spicyz to determine the version.
+                spicyz = find_program("spicyz")
+                if not spicyz:
+                    LOG.warning(
+                        "Builtin Zeek::Spicy plugin found, but spicyz not in path"
+                    )
+
+                    # Use Zeek's version if there's no spicyz around.
+                    spicy_version = get_zeek_version()
+                else:
+                    spicy_version = subprocess.check_output(
+                        [spicyz, "--version"], timeout=10
+                    ).decode("utf-8")
+
+                return self._mangle_plugin_version(spicy_version)
+
+        return None
+
+    def discover_builtin_plugins(self):
+        """
+        Discover Zeek builtin plugins and their versions for dependency resolution.
+
+        This is using the `--build-info` flag which was introduced with Zeek 6.0.
+
+        Returns:
+            List of BuiltinPlugin instances with name, version and info
+            accessors.
+        """
+
+        class BuiltinPlugin(typing.NamedTuple):
+            name: str
+            version: str
+            info: PackageInfo
+
+            @classmethod
+            def make(cls, name, version):
+                package = Package(git_url=f"builtin://{name}")
+                status = PackageStatus(
+                    tracking_method=TRACKING_METHOD_VERSION,
+                    current_version=version,
+                )
+                info = PackageInfo(
+                    package=package,
+                    status=status,
+                    versions=[version],
+                )
+                return cls(name, version, info)
+
+        if self._builtin_plugins is not None:
+            return self._builtin_plugins
+
+        self._builtin_plugins = []
+        zeek = find_program("zeek")
```

Comment:
This is a-okay — I'm just wondering whether we should centralize Zeek-related invocations somehow. zkg already looks up `zeek-config` in several settings. There's `_util.get_zeek_version()`, which feels a bit ad-hoc too. Could do a singleton summarizing the Zeek installation, for example. (Could definitely be a separate PR.)